### PR TITLE
Update Jquery.validation.Unobtrusive

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Part of https://github.com/aspnet/Templating/issues/32.

Things I also did while I was at it:

- Remove referenced to Bower (same as we're doing in templating).
- Remove site.min.* files (same as we did in templating).
- Move to CDNJS from https://ajax.aspnetcdn.com (same as we did in templating).

@javiercn let me know if any of this needs to remain unchanged do to legacy. For example, should we just revert all the changes I made in */V3?

### CODE REVIEW

File: src/UI/Areas/Identity/Pages/V3/_Layout.cshtml [hunk 1]
```diff
@@ -14,10 +14,10 @@
         <link rel="stylesheet" href="~/Identity/css/site.css" />
     </environment>
     <environment exclude="Development">
-        <link rel="stylesheet" href="https://ajax.aspnetcdn.com/ajax/bootstrap/3.3.7/css/bootstrap.min.css"
+        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css"
               asp-fallback-href="~/Identity/lib/bootstrap/dist/css/bootstrap.min.css"
               asp-fallback-test-class="sr-only" asp-fallback-test-property="position" asp-fallback-test-value="absolute" />
-        <link rel="stylesheet" href="~/Identity/css/site.min.css" asp-append-version="true" />
+        <link rel="stylesheet" href="~/Identity/css/site.css" asp-append-version="true" />
```
Comment (javiercn) [inline src/UI/Areas/Identity/Pages/V3/_Layout.cshtml]: This strikes me as wrong. The CSS inside the production library should be the minified version of the code.

File: src/UI/wwwroot/V4/Identity/lib/jquery-validation-unobtrusive/jquery.validate.unobtrusive.js [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
+// @version v3.2.11
 
 /*jslint white: true, browser: true, onevar: true, undef: true, nomen: true, eqeqeq: true, plusplus: true, bitwise: true, regexp: true, newcap: true, immed: true, strict: false */
 /*global document: false, jQuery: false */
 
 (function (factory) {
     if (typeof define === 'function' && define.amd) {
         // AMD. Register as an anonymous module.
-        define("jquery.validate.unobtrusive", ['jquery.validation'], factory);
+        define("jquery.validate.unobtrusive", ['jquery-validation'], factory);
```
Comment (javiercn) [inline src/UI/wwwroot/V4/Identity/lib/jquery-validation-unobtrusive/jquery.validate.unobtrusive.js]: Is this intentional? This is changing the dependency on the define

Comment (ryanbrandenburg) [reply src/UI/wwwroot/V4/Identity/lib/jquery-validation-unobtrusive/jquery.validate.unobtrusive.js]: https://github.com/aspnet/jquery-validation-unobtrusive/issues/79. Intentional, this copy was just more than one version out of date.

File: src/UI/Areas/Identity/Pages/V3/_Layout.cshtml (hunk 1 above)
Comment (ryanbrandenburg) [reply src/UI/Areas/Identity/Pages/V3/_Layout.cshtml]: We discussed this [here](https://github.com/aspnet/Templating/pull/729). We decided that without an out-of-the-box way to minify site.min.* on build it was more dangerous than it was helpful to include those files which can quickly get out of sync with their non-minified versions.

In an update I've removed the site.min.* files, the same as we did in templating.

Comment (javiercn) [reply src/UI/Areas/Identity/Pages/V3/_Layout.cshtml]: Sure

**Comment by javiercn** [comment]
What manual testing have you done on these changes?

**Comment by ryanbrandenburg** [comment]
To test I launched test/WebSites/Identity.DefaultUI.WebSite (bootstrap4) and test/WebSites/Identity.OpenIdConnect.WebSite (bootstrap3), confirmed that they had no errors in the console, looked about right, and that when you tried to submit an empty form you got the appropriate warnings in the UI.

**Comment by javiercn** [comment]
@ryanbrandenburg You need to run the app in development and in production, launching it (I assume with F5 or dotnet run) is not good enough.
We don't care about `test/WebSites/Identity.OpenIdConnect.WebSite` but for the default UI you need to run it on both environments as described above and make sure that there are no errors on the browser console (Visit login/register/manage)

**Comment by ryanbrandenburg** [comment]
I assume by that you mean:

- `dotnet run -f netcoreapp2.2 -c Release`
- `dotnet run -f netcoreapp2.2 -c Debug`

If so I've now tested it that way and we get no errors and working pages both ways.

**Comment by ryanbrandenburg** [comment]
If I do that I get:
```
Unhandled Exception: System.IO.FileNotFoundException: Related assembly 'Microsoft.AspNetCore.Identity.UI.Views.V3' specified by assembly 'Microsoft.AspNetCore.Identity.UI' could not be found in the directory D:\dd\Identity\test\WebSites\Identity.DefaultUI.WebSite\bin\Release\netcoreapp2.2\publish. Related assemblies must be co-located with the specifying assemblies.
   at Microsoft.AspNetCore.Mvc.ApplicationParts.RelatedAssemblyAttribute.GetRelatedAssemblies(Assembly assembly, Boolean throwOnError, Func`2 fileExists, Func`2 loadFile)
   at Microsoft.AspNetCore.Mvc.ApplicationParts.RelatedAssemblyAttribute.GetRelatedAssemblies(Assembly assembly, Boolean throwOnError)
   at Microsoft.AspNetCore.Identity.IdentityBuilderUIExtensions.<>c__DisplayClass3_0.<AddRelatedParts>b__0(ApplicationPartManager partManager) in D:\dd\Identity\src\UI\IdentityBuilderUIExtensions.cs:line 77
   at Microsoft.Extensions.DependencyInjection.MvcCoreMvcBuilderExtensions.ConfigureApplicationPartManager(IMvcBuilder builder, Action`1 setupAction)
   at Microsoft.AspNetCore.Identity.IdentityBuilderUIExtensions.AddRelatedParts(IdentityBuilder builder, UIFramework framework) in D:\dd\Identity\src\UI\IdentityBuilderUIExtensions.cs:line 132
   at Microsoft.AspNetCore.Identity.IdentityBuilderUIExtensions.AddDefaultUI(IdentityBuilder builder, UIFramework framework) in D:\dd\Identity\src\UI\IdentityBuilderUIExtensions.cs:line 54
   at Microsoft.AspNetCore.Identity.IdentityBuilderUIExtensions.AddDefaultUI(IdentityBuilder builder) in D:\dd\Identity\src\UI\IdentityBuilderUIExtensions.cs:line 33
   at Microsoft.Extensions.DependencyInjection.IdentityServiceCollectionUIExtensions.AddDefaultIdentity[TUser](IServiceCollection services, Action`1 configureOptions) in D:\dd\Identity\src\UI\IdentityServiceCollectionUIExtensions.cs:line 56
   at Microsoft.Extensions.DependencyInjection.IdentityServiceCollectionUIExtensions.AddDefaultIdentity[TUser](IServiceCollection services) in D:\dd\Identity\src\UI\IdentityServiceCollectionUIExtensions.cs:line 33
   at Identity.DefaultUI.WebSite.StartupBase`2.ConfigureServices(IServiceCollection services) in D:\dd\Identity\test\WebSites\Identity.DefaultUI.WebSite\StartupBase.cs:line 41
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Hosting.ConventionBasedStartup.ConfigureServices(IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.EnsureApplicationServices()
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.Initialize()
   at Microsoft.AspNetCore.Hosting.WebHostBuilder.Build()
   at Identity.DefaultUI.WebSite.Program.Main(String[] args) in D:\dd\Identity\test\WebSites\Identity.DefaultUI.WebSite\Program.cs:line 16
```

which is a pre-existing condition because it happens on a totally clean branch.

By copying those dll's into the publish directory I found a couple instances of site.min.* remaining, which I fixed.

**Comment by javiercn** [comment]
@ryanbrandenburg Those issues are expected as we are referencing the assembly as a project reference and not a package and we probably didn’t fixup publish.

Have you tried both versions of the UI? (V3 and V4)

**Comment by ryanbrandenburg** [comment]
Fixed one extra item I found when testing in V3.

**Comment by ryanbrandenburg** [comment]
Is that approval then?

**Pull request merged**